M2/Macaulay2/e/M2FreeAlgebraQuotient.cpp

```diff
@@ -0,0 +1,378 @@
+#include "M2FreeAlgebraQuotient.hpp"
+
+#include <iostream>
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "NCAlgebras/FreeAlgebra.hpp"
+#include "matrix.hpp"
+#include "ring.hpp"
+#include "stdinc-m2.hpp"
```

Comment:
What is `stdinc-m2.hpp`? I forgot to ask this when I was merging the code. Should it go in `engine-includes.hpp`?